### PR TITLE
feat: pnpm-style symlinking (#636)

### PR DESCRIPTION
**Pull request by netlify-team-account-1**
Contains a reproduction test for pnpm-style symlinking, reported non-working in #636.

### CODE REVIEW

**Comment by eduardoboucas** [comment]
@netlify-team-account-1 what's the status of this one?

**Comment by netlify-team-account-1** [comment]
This is the test that was originally added a while ago to reproduce #636 (the issue we paired on). We found that this reproduction doesn't correctly capture the real issue - this test may still be useful, so why not merge it.

**Comment by eduardoboucas** [comment]
> This is the test that was originally added a while ago to reproduce #636 (the issue we paired on). We found that this reproduction doesn't correctly capture the real issue - but the test may still be useful to prevent future bugs, so why not merge it.

Sounds good!

File: tests/main.js
Comment (eduardoboucas) [inline tests/main.js:1985]: We'll need to remove these once we land https://github.com/netlify/zip-it-and-ship-it/pull/682.

**Review by eduardoboucas — APPROVED** [review]

File: tests/main.js
Comment (netlify-team-account-1) [reply tests/main.js:1985]: #682 is only about fixtures, not about test code. We could add `// eslint-disable-file import/no-dynamic-require, node/global-require` to the top of this file instead of line-by-line, though!

**Comment by erlandsona** [comment]
Curious if this also solves for this issue with regular `npm`?

**Comment by Skn0tt** [comment]
Not sure what issue you're referring to - if there's already an issue for it in this repository, could you link it? If not, feel free to open one :)

**Pull request merged**
